### PR TITLE
node:*: fix some typescript errors

### PR DESCRIPTION
**Pull request by nektro**
notable fixes

- missing validators
- missing primordials
- primordials defined wrong
- missing identifiers
- missing builtins
- missing type definitions
- duplicate definitions clobbering names
- `ArrayIsArray` being called wrong
- passing arguments to functions that don't use them
- misc parameter and array type safety
- error builtins being called wrong

### CODE REVIEW

File: src/js/node/child_process.ts [hunk 1]
```diff
@@ -1621,6 +1621,10 @@ function validateString(value, name) {
   if (typeof value !== "string") throw new ERR_INVALID_ARG_TYPE(name, "string", value);
 }
 
+function isInt32(value) {
```
Comment (Jarred-Sumner) [inline src/js/node/child_process.ts:1623]: is this a builtin? I think it's a builtin

File: src/js/node/diagnostics_channel.ts [hunk 2]
```diff
@@ -3,10 +3,10 @@
 const SafeMap = Map;
 const SafeFinalizationRegistry = FinalizationRegistry;
 
-const ArrayPrototypeAt = (array, index) => array[index];
-const ArrayPrototypeIndexOf = (array, value) => array.indexOf(value);
-const ArrayPrototypePush = (array, value) => array.push(value);
-const ArrayPrototypeSplice = (array, start, deleteCount) => array.splice(start, deleteCount);
+const ArrayPrototypeAt = Array.prototype.at;
+const ArrayPrototypeIndexOf = Array.prototype.indexOf;
+const ArrayPrototypePush = Array.prototype.push;
```
Comment (Jarred-Sumner) [inline src/js/node/diagnostics_channel.ts]: most of these should actually be `$arrayPush(array, value)`

File: src/js/node/readline.ts [hunk 3]
```diff
@@ -382,7 +386,7 @@ function validateObject(value, name, options = null) {
   var nullable = options?.nullable ?? false;
   if (
     (!nullable && value === null) ||
-    (!allowArray && ArrayIsArray.$call(value)) ||
+    (!allowArray && ArrayIsArray.$call(null, value)) ||
```
Comment (Jarred-Sumner) [inline src/js/node/readline.ts]:
```suggestion
    (!allowArray && $isJSArray(value)) ||
```

File: src/js/node/child_process.ts (hunk 1 above)
Comment (nektro) [reply src/js/node/child_process.ts:1623]: node src: https://github.com/nodejs/node/blob/c1d9f426a11eda97/lib/internal/validators.js#L39-L45

**Pull request merged**